### DISCUSSION TITLE
Looking for a Reference on the Report on the CLI

### DISCUSSION THREAD

**Discussion by eliezercazares** [artilleryio/artillery#1460] [Artillery]
Hello,
Say I want to know how many users my application supports...

I have changed the arrivalRate and rampTo values for bigger, and of course the chat application breaks earlier with less values...

My question is, how can I determine the concurrence threshold approximately? 

I add my test and the summary below...

Thanks all!
Great job here.

`config:
  target: "ws://localhost:50000"
  ensure:
    max: 2000 # fail if max response time exceeds x ms
    maxErrorRate: 1 # fail if error rate exceeds 1%
  socketio:
    transports: ["websocket"]
  processor: "./custom.js" # set a custom processor to load dynamic scenario data
  phases:
    - duration: 300 # Run phase for s seconds
      arrivalRate: 5 # Create x virtual users per second
      rampTo: 15
      name: "Loading users per second and ramping up their numbers for x seconds"
scenarios:
  - engine: "socketio"
    flow:
      - function: "getSetProfileData" # load variables
      - emit:
          channel: "setProfile"
          data: "{{ setProfileData }}"
      - think: 10 # do nothing for x seconds, then disconnect
      - emit:
          channel: "rejoinPlace"
          data: "{{ rejoinPlace }}"
      - think: 5 # do nothing for x seconds, then disconnect
      - loop:
        - emit:
            channel: "messageSent"
            data: "{{ messageSent }}"
        - think: 5 # do nothing for x seconds, then disconnect
        - emit:
            channel: "messageReply"
            data: "{{ messageSentReply }}"
        - think: 5 # do nothing for x seconds, then disconnect
      count: 20`

Summary with 273 MB for nodes process In chat app
`
errors.Error: websocket error: ................................................. 1488
socketio.emit: ................................................................. 2864
socketio.emit_rate: ............................................................ 10/sec
socketio.response_time:
  min: ......................................................................... 0
  max: ......................................................................... 1
  median: ...................................................................... 0.1
  p95: ......................................................................... 0.2
  p99: ......................................................................... 0.3
vusers.completed: .............................................................. 603
vusers.created: ................................................................ 2091
vusers.created_by_name.0: ...................................................... 2091
vusers.failed: ................................................................. 1488
vusers.session_length:
  min: ......................................................................... 25002.7
  max: ......................................................................... 25590.1
  median: ...................................................................... 25091.6
  p95: ......................................................................... 25598.5
  p99: ......................................................................... 25598.5
`

Summary with 2gb for the process of the chat app:
`------------------------------

errors.Error: websocket error: ................................................. 163
socketio.emit: ................................................................. 7884
socketio.emit_rate: ............................................................ 23/sec
socketio.response_time:
  min: ......................................................................... 0
  max: ......................................................................... 2.1
  median: ...................................................................... 0.1
  p95: ......................................................................... 0.2
  p99: ......................................................................... 0.3
vusers.completed: .............................................................. 1971
vusers.created: ................................................................ 2134
vusers.created_by_name.0: ...................................................... 2134
vusers.failed: ................................................................. 163
vusers.session_length:
  min: ......................................................................... 25001.9
  max: ......................................................................... 39923.5
  median: ...................................................................... 25598.5
  p95: ......................................................................... 32542.3
  p99: ......................................................................... 39747.5
`

**Comment by GeekyDeaks** [comment]
I ended up making a post-run script to approximate the concurrent users and add them as a metric to the report.  I mentioned it in an earlier discussion on the topic:

https://github.com/artilleryio/artillery/discussions/1449#discussioncomment-2773312
